core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -892,16 +889,11 @@ class UnifiedLog(@volatile var logStartOffset: Long,
 
           maybeDuplicate match {
             case Some(duplicate) =>
-              appendInfo.setFirstOffset(Optional.of(new LogOffsetMetadata(duplicate.firstOffset)))
+              appendInfo.setFirstOffset(duplicate.firstOffset)
               appendInfo.setLastOffset(duplicate.lastOffset)
               appendInfo.setLogAppendTime(duplicate.timestamp)
               appendInfo.setLogStartOffset(logStartOffset)
             case None =>
-              // Before appending update the first offset metadata to include segment information
-              appendInfo.setFirstOffset(appendInfo.firstOffset.map { offsetMetadata =>
```

Comment:
We removed this since the default is zero, and in that case we take the segment base offset anyway?

Reply:
We removed this because none of the readers use anything besides the `messageOffset` (which is the `appendInfo.firstOffset` now).

Reply:
Right -- was just clarifying we use the default.